tests/ui/traits/non_lifetime_binders/nested-apit-mentioning-outer-bound-var.rs

```diff
@@ -0,0 +1,11 @@
+#![feature(non_lifetime_binders)]
+//~^ WARN the feature `non_lifetime_binders` is incomplete
+
+trait Trait<Input> {
+    type Assoc;
+}
+
+fn uwu(_: impl for<T> Trait<(), Assoc = impl Trait<T>>) {}
```

Comment:
Huh, I didn't even know `for<T>` was a thing!

Reply:
https://github.com/rust-lang/rust/pull/107489